src/Encoding/FilteredStream.php

```diff
@@ -57,10 +57,22 @@
      */
     public function __construct(StreamInterface $stream, $readFilterOptions = null, $writeFilterOptions = null)
     {
+        $triggerDeprecationNotice = null !== $writeFilterOptions;
+
+        if (null === $readFilterOptions && in_array($this->readFilter(), ['convert.base64-encode', 'convert.base64-decode', 'convert.quoted-printable-encode', 'zlib.deflate', 'zlib.inflate'/*, 'bzip2.compress', 'bzip2.decompress'*/], true)) {
```

Comment:
Unfortunately the code as you wrote it won't work because `Filter\fun` by default pass `null` as value of the `$params` argument of the `stream_filter_append` function. I proposed a solution in #78 that I tested with the `base64_encode` filter (which is the reason I found the problem and opened the issue), but @dbu told me to explicitly check for the filter names. I don't know if there is a third solution, but at the moment the only two that I tested and works are these